portality/formcontext/formcontext.py

```diff
@@ -797,8 +797,8 @@ def finalise(self):
             raise FormContextException(
                 "You don't have permission to edit applications which are in status {0}.".format(src_status))
 
-        # Don't permit changes to status in reverse of the editorial process
-        if editor_choices.index(target_status) < editor_choices.index(src_status):
+        # Don't permit changes to status in reverse of the editorial process for statuses other than 'in progress'
+        if target_status != 'in progress' and editor_choices.index(target_status) < editor_choices.index(src_status):
```

Comment:
I have to confess I'm not a huge fan of this "forward only, except in these conditions".  It would be nicer to have a function which would give you your allowed statuses given your current status, which could be done with the Choices object, perhaps.  Something like:

editors_choices = choices.Choices.application_status("editor", mycurrentstatus)

I realise that this is just one example, and this file is probably full of situations where we could better encapsulate the workflow.